File: Python/string_array_mem.py
import time

ONE_MILLION = 1000000
ONE_BILLION = 1000000000
NUM_TRIALS = 50
cumulative = 0

x = 0
strings = []
while x < 100000:
    if x == 0:
        strings.append("a")
    else:
       strings.append(strings[x-1] + "a")
    x += 1


input()
    

File: Rust/string_array_mem.rs
use std::io;

fn main() {
    const ONE_BILLION_D : f64 = 1000000000.0;
    const ONE_BILLION : u64 = 1000000000;
    const ONE_MILLION : usize = 1000000;
    const NUM_TRIALS : u16 = 50;

    let mut x : usize = 0;
    let mut strings: Vec<String> = Vec::with_capacity(100000);
    while x < 100000 {
        if x == 0 {
            strings.push("a".to_string());
        } else {
            strings.push(strings[x - 1].clone() + "a")
        }
        x += 1;
    }

    let mut buffer = String::new();
    io::stdin().read_line(&mut buffer);

}